### ISSUE TITLE
Install using http://server.com:port overwrites port# with 2368

### ISSUE THREAD

**Issue by awbacker** [TryGhost/Ghost-CLI#519]
When installing and using a URL with a port in it, the port is seen but the value is replaced with 2368 in both the URL and the `port` setting in the config.  

Steps performed:
----
* Create directory `/opt/ghost-v1/`, and `chown ghost:ghost` that thing
* Run ghost install in the directory as root
* Enter "https://somesite.co:4000" as the url

Results:
----
* Config contains `url = https://somesite.co:4000`
* Config contains `port = 2368`

Desired Results:
----
Not really sure what is the right/easy/common way to go.  It would be nice to have 2 settings, one for port and one for url, though I could just edit the config file after and remove the `:4000` from the blog url, since its fronted by nginx.  But if its there they should at least be the same.

Install Output:
----
```bash
root@linode:/opt/ghost-v1# ghost install --db=sqlite3
✔ Checking system Node.js version
✔ Checking current folder permissions
...
✔ Downloading and installing Ghost v1.16.2
✔ Finishing install process

? Enter your blog URL: http://somesite.co:4000
✔ Configuring Ghost
✔ Setting up instance
...
? Do you wish to set up Systemd? No
ℹ Setting up Systemd [skipped]
Running sudo command: /usr/lib/node_modules/ghost-cli/node_modules/.bin/knex-migrator-migrate --init --mgpath /opt/ghost-v1/current
...
```

Resulted in this config (unrelated sections omitted)
```
root@linode:/opt/ghost-v1# cat config.production.json
{
  "url": "http://somesite.co:2368/",
  "server": {
    "port": 2368,
    "host": "127.0.0.1"
  },
 ....
```

**Comment by acburdine** [comment]
Hmm...so it _should_ be taking the port that you specified in the url and using it. What _used_ to happen was if you specified somesite.com:4000, it would set the server.port value to 4000.

This is likely a bug, I'll do some more investigating.

**Comment by NistorCristian** [comment]
I can confirm this too. It also breaks the install if you have multiple Ghost instances:

```
✖ Configuring Ghost
A ConfigError occurred.

Error detected in the undefined configuration.

Message: Port '2368' is in use.
Configuration Key(s): server.port
Current Value(s): 2368
```

To solve this I run `ghost config --port 2371` (2371 is a port that isn't used anywhere)
I also put the port in url:
```
Enter your blog URL: https://blog.domain.com:2371
etc...
```

Then the config file is created.

I removed the port from `config.production.json` url.
`"url": "https://blog.domain.com",` became `"url": "https://blog.domain.com",`

After this I run manually all these commands:
https://docs.ghost.org/v1/docs/cli-setup#section-arguments

I managed to install the blog and it works. I only see a problem that I don't know why this happens. Maybe @acburdine knows why.

When I run `ghost ls` the new blog doesn't appear in the list. Any ideas?
